projects/igniteui-angular/src/lib/select/select-positioning-strategy.ts

```diff
@@ -31,201 +25,231 @@ export class SelectPositioningStrategy extends ConnectedPositioningStrategy impl
+    private compensateYScroll(selectFit: SelectFit, compensation: number ) {
+        selectFit.verticalOffset += compensation;
+        this.global_yOffset = selectFit.verticalOffset;
+    }
+
+    // Position the items outer container Below or Above the input.
+    fitInViewport(contentElement: HTMLElement, selectFit: SelectFit) {
+        // Position Select component's container below target/input as preferred positioning over above target/input
+        if (this.canFitBelowInput(selectFit) || !this.canFitAboveInput(selectFit)) {
+                // Calculate container starting point;
+                // TODO: modify the yOffset instead & use one call to super.setStyle
+                selectFit.top = selectFit.targetRect.top - selectFit.styles.itemTextToInputTextDiff;
+
+        } else {
+            // Position Select component's container above target/input
+            // TODO: modify the yOffset instead & use one call to super.setStyle
+            selectFit.top = selectFit.targetRect.bottom + selectFit.styles.itemTextToInputTextDiff -
+                selectFit.contentElementRect.height;
+            }
+    }
 
+    protected canFitBelowInput(selectFit: SelectFit): boolean {
+        return selectFit.targetRect.top - selectFit.styles.itemTextToInputTextDiff + selectFit.contentElementRect.height <
+        selectFit.viewPortRect.bottom;
+    }
+
+    protected canFitAboveInput(selectFit: SelectFit): boolean {
+            return selectFit.targetRect.bottom + selectFit.styles.itemTextToInputTextDiff -
+            selectFit.contentElementRect.height > selectFit.viewPortRect.top;
+    }
+
+    protected setStyles(contentElement: HTMLElement, selectFit: SelectFit, initialCall?: boolean) {
+        // The Select component's container is about to be displayed. Set its position.
+        if (initialCall) {
+            contentElement.style.top = `${selectFit.top}px`;
+            contentElement.style.left = `${selectFit.left}px`;
+        // Page's View Window container is scrolled. Reposition Select's container.
+        } else {
+            super.setStyle(contentElement, selectFit.targetRect, selectFit.contentElementRect, selectFit);
+        }
+
+        contentElement.style.width = `${selectFit.styles.contentElementNewWidth}px`; // manage container based on paddings?
+        this.global_styles.contentElementNewWidth = selectFit.styles.contentElementNewWidth;
+    }
+
+    private calculateVariables(selectFit: SelectFit) {
+        selectFit.itemRect = this.getInteractionItemElement().getBoundingClientRect();
+        selectFit.itemElement = this.getInteractionItemElement();
+        selectFit.dropDownList = this.select.scrollContainer;
+        selectFit.inputElement = this.select.getEditElement();
+    }
+
+    public calculateStyles(selectFit: SelectFit): SelectStyles  {
+        const styles: SelectStyles = {};
+        const inputFontSize = window.getComputedStyle(selectFit.inputElement).fontSize;
```

Comment:
Cache what `getComputetStyle` returns and call it once on each element needed.
Now you are calling it three times for `inputElement` and another three times for `itemElement`

Reply:
done